src/main/java/cc/woverflow/hytils/mixin/GuiIngameForgeMixin_HideHotbar.java

```diff
@@ -0,0 +1,201 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package cc.woverflow.hytils.mixin;
+
+import cc.polyfrost.oneconfig.utils.hypixel.HypixelUtils;
+import cc.polyfrost.oneconfig.utils.hypixel.LocrawInfo;
+import cc.polyfrost.oneconfig.utils.hypixel.LocrawUtil;
+import cc.woverflow.hytils.HytilsReborn;
+import cc.woverflow.hytils.config.HytilsConfig;
+import net.minecraft.client.Minecraft;
+import net.minecraftforge.client.GuiIngameForge;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Arrays;
+
+@Mixin(value = GuiIngameForge.class, remap = false)
+public class GuiIngameForgeMixin_HideHotbar {
+    // currently doesn't account for admin abuse lobbies, duels lobby pvp, or housing pvp
+    @Inject(method = "renderHealth", at = @At("HEAD"), cancellable = true)
+    public void cancelHealthbar(int width, int height, CallbackInfo ci) {
+        LocrawInfo locraw = LocrawUtil.INSTANCE.getLocrawInfo();
+        if (HytilsConfig.hideHudElements && locraw != null && HypixelUtils.INSTANCE.isHypixel()) {
+            if (HytilsReborn.INSTANCE.getLobbyChecker().playerIsInLobby()) {
+                // rudimentary check if player has engaged in pvp or something
+                if (Minecraft.getMinecraft().thePlayer.getHealth() < 20) {
+                    return;
+                }
+                ci.cancel();
+                return;
+            }
+
+            LocrawInfo.GameType gameType = locraw.getGameType();
+            String gameMode = locraw.getGameMode();
+            switch (gameType) {
+                case DROPPER:
+                case HOUSING:
+                    // rudimentary check if player has engaged in pvp or something
+                    if (Minecraft.getMinecraft().thePlayer.getHealth() < 20) {
+                        break;
+                    }
+                case BUILD_BATTLE:
+                case LIMBO:
+                case QUAKE:
+                case TNT_GAMES:
+                    // break out of the switch if it's a duels game that has varying health
+                    if (gameMode.contains("CAPTURE") || gameMode.contains("PVPRUN")) {
+                        break;
+                    }
+                case REPLAY:
+                    ci.cancel();
+                    return;
+            }
+
+            Arrays.asList(
```

Comment:
can you use a switch statement or at least like not create the list every frame lol

Reply:
Fixed in my PR, merge mine first to the hide-hud branch